### ISSUE TITLE
geoJson: layers based on feature property

### ISSUE THREAD

**Issue by MareinK** [Leaflet/Leaflet#8334]
### Checklist

- [X] I've searched through the [plugins](https://leafletjs.com/plugins.html) to make sure this feature isn't already available, or think it shouldn't require a plugin.
- [X] I've searched through the [current issues](https://github.com/Leaflet/Leaflet/issues) to make sure this feature hasn't been requested already.
- [X] I agree to follow the [Code of Conduct](https://github.com/Leaflet/Leaflet/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.

### Motivation

I have a geoJson file containing all sorts of features. I would like to have these features grouped into different layers in Leaflet.

### Suggested solution

I think it would be elegant to be able to specify, in the geoJson call, a property of the features which indicates the name of the layer that the property should belong to. Leaflet itself would then sort out which layers should exist and assign the features accordingly.

### Alternatives considered

I know one way I could do this, is by separating the features into multiple files, but I don't find that sustainable as the number of layers grows, and moving a feature from one layer to another is also tedious. Additionally I need to change the code whenever I add/remove layers, to include/exclude the new/old file.

I could also keep all the features in one file and give them the suggested `layer` property, then first extract the set of those and then repeatedly use Leaflet's geoJson 'filter' functionality to get the result I want.

Basically this but more dynamic: https://stackoverflow.com/a/37039655

But it would be nice if Leaflet offers this functionality out-of-the-box.

Thanks for any insights!
